src/zope/app/apidoc/static.py

```diff
@@ -115,108 +121,156 @@ def isLocalURL(self):
 
     def isApidocLink(self):
         # Make sure that only apidoc links are loaded
-        if self.absoluteURL.startswith(self.rootURL+'++apidoc++/'):
-            return True
-        if self.absoluteURL.startswith(self.rootURL+'@@/'):
+        allowed_prefixes = ((self.rootURL + '++apidoc++/'),
+                            (self.rootURL + '@@/'))
+        if self.absoluteURL.startswith(allowed_prefixes):
             return True
-        return False
-
 
-class OnlineBrowser(mechanize.Browser, object):
 
-    def __init__(self, factory=None, history=None, request_class=None):
-        if factory == None:
-            factory = ApiDocDefaultFactory()
-        mechanize.Browser.__init__(self, factory, history, request_class)
+class OnlineBrowser(zope.testbrowser.browser.Browser):
 
     def setUserAndPassword(self, user, pw):
         """Specify the username and password to use for the retrieval."""
-        hash = base64.encodestring(user+':'+pw).strip()
-        self.addheaders.append(('Authorization', 'Basic '+hash))
+        user_pw = user + ':' + pw
+        if not isinstance(user_pw, bytes):
+            user_pw = user_pw.encode('latin-1')
```

Comment:
For my future reference, since I googled it up: https://tools.ietf.org/html/rfc7617#page-5

TL;DR: the password charset is undefined; some clients use Latin-1, some use UTF-8, some use locale charset. The server can ask for a specific charset in the basic auth challenge, provided that the charset it asks for is UTF-8.  (Literally, "The only allowed value is "UTF-8"".  Oh, and the password's supposed to be converted to Normalization Form C.)